packages/grafana-ui/src/components/Logs/LogDetails.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { PureComponent } from 'react';
+import { LogsParser, getParser, LogRowModel } from '@grafana/data';
+
+import { Themeable } from '../../types/theme';
+import { withTheme } from '../../themes/index';
+import { getLogRowStyles } from './getLogRowStyles';
+
+//Components
+import { LogDetailsRow } from './LogDetailsRow';
+
+interface Props extends Themeable {
+  row: LogRowModel;
+  getRows: () => LogRowModel[];
+  onClickFilterLabel?: (key: string, value: string) => void;
+  onClickFilterOutLabel?: (key: string, value: string) => void;
+}
+
+interface State {
+  parsed: boolean;
+  parsedFields: string[];
+  parser?: LogsParser;
+}
+
+class UnThemedLogDetails extends PureComponent<Props, State> {
+  state: State = {
+    parsed: false,
+    parser: undefined,
+    parsedFields: [],
+  };
+
+  parseMessage() {
```

Comment:
hmm I would probably memoize this and do it during render, otherwise you should be doing this also if row.entry was updated from new props.